aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -1,159 +1,72 @@
 #include <ATen/native/TensorTransformations.h>
-#include <ATen/WrapDimUtilsMulti.h>
 
 #include <ATen/NativeFunctions.h>
 #include <ATen/Parallel.h>
+#include <ATen/WrapDimUtilsMulti.h>
+#include <ATen/core/DimVector.h>
 #include <c10/util/Exception.h>
 
 #include <algorithm>
 #include <vector>
 
 namespace at {
 namespace native {
-namespace {
-
-constexpr size_t dim_bitset_size = 64;
-
-Tensor build_index(Tensor input, int64_t flip_dim) {
-  int64_t element_size_bytes = input.element_size();
-  auto num_dims = input.ndimension();
-  auto dim_size = input.size(flip_dim);
-  auto stride = input.stride(flip_dim);
-
-  auto new_shape = std::vector<int64_t>(num_dims, 1);
-  new_shape[flip_dim] = dim_size;
 
-  TensorOptions tensor_options =
-    TensorOptions(c10::kLong).
-    device(c10::kCPU);
-
-  auto index = at::empty(new_shape, tensor_options);
-  auto input_index_ptr = index.data_ptr<int64_t>();
-
-  for(int64_t i = 0; i < dim_size; i++) {
-    input_index_ptr[i] = static_cast<int64_t>(dim_size - i - 1) * stride * element_size_bytes;
+Tensor flip(const Tensor& self, IntArrayRef dims) {
+  auto in_tensor = self;
+  const int64_t total_dims = in_tensor.dim();
+  auto flip_dims_b = at::dim_list_to_bitset(dims, total_dims);
+  Tensor out_tensor = at::empty_like(in_tensor, MemoryFormat::Contiguous);
```

Comment:
please don't hardcode output layout here, generally operations try to propagate the permutation of the input tensor.

Reply:
Oh, what's the reason for this? I thought that contiguous was always better.

Reply:
Makes sense, thanks!